OpenSim/Common/Component.cpp

```diff
@@ -202,19 +208,25 @@ void Component::connect(Component &root)
 
 
 // Call connect on all components and find unconnected Connectors a
-void Component::componentsConnect(Component& root) const
+void Component::componentsConnect(Component& root) 
 {
     // First give the subcomponents the opportunity to connect themselves
-    for(unsigned int i=0; i<_components.size(); i++){
-        _components[i]->connect(root);
+    for (unsigned int i = 0; i<_memberSubcomponents.size(); ++i) {
+        _memberSubcomponents[i].upd()->connect(root);
+    }
+    for(unsigned int i=0; i<_propertySubcomponents.size(); ++i){
+        _propertySubcomponents[i].get()->connect(root);
+    }
+    for (unsigned int i = 0; i<_adoptedSubcomponents.size(); ++i) {
+        _adoptedSubcomponents[i].upd()->connect(root);
     }
 }
 
 void Component::disconnect()
 {
     // First give the subcomponents the opportunity to disconnect themselves
-    for (unsigned int i = 0; i<_components.size(); i++){
-        _components[i]->disconnect();
+    for (unsigned int i = 0; i<_propertySubcomponents.size(); i++){
```

Comment:
Thinking about it more, I think you are right and I should disconnect all subcomponents for completeness. Model building/editing in the GUI could require that all connectors of a Component (including its subs) must be disconnected before removing/deleting, for example.